Telemetry module (+ sonification) documentation (#37)
OpenSpace/OpenSpace-Docs: reference/modules/telemetry/specialized.md

  2. The horizontal angle in radians to the planet, with the current angle calculation mode taken into account.
  3. The elevation angle in radians to the planet, with the current angle calculation mode taken into account.
  4. List of user interface settings for the planet sonification, which aspectes of the sonification should be turned on or off. A value of 0 means that it is turned off and a 1 means that it is turned on. The order of the settings can be seen in the table below. If the setting does not exist for a planet, the value is always 0.
  5. The distance from the camera to the first moon in kilometers
  6. The horizontal angle in radians to the first moon, with the current angle calculation mode taken into account.
  7. The elevation angle in radians to the first moon, with the current angle calculation mode taken into account.
  8. The distance from the camera to the second moon in kilometers
  9. The horizontal angle in radians to the second moon, with the current angle calculation mode taken into account.
  10. The elevation angle in radians to the second moon, with the current angle calculation mode taken into account.

>   11. Data continues in the same pattern for as many moons there are for the planet

Author: I tried to make it a little more readable now, but I would like to avoid the sub-lists. I think this could be interpreted as one message index containing three items (essentially and array in an array slot), which is not the case here.

- `2802296365.598`: The distance from the camera to the fourth moon, Titania, in kilometers.
- `0.18863273887734`: The horizontal angle in radians to the fourth moon, Titania. In this example the angle calculation mode Circular was used with the elevation angle included.
- `2.195427733976`: The elevation angle in radians to the fourth moon, Titania. In this example the angle calculation mode Circular was used with the elevation angle included.
- `2802217828.1715`: The distance from the camera to the fifth moon, Umbriel, in kilometers.
- `0.99476981366003`: The horizontal angle in radians to the fifth moon, Umbriel. In this example the angle calculation mode Circular was used with the elevation angle included.
- `2.0118520154497`: The elevation angle in radians to the fifth moon, Umbriel. In this example the angle calculation mode Circular was used with the elevation angle included.
:::

::::

## Planets Compare Sonification
This telemetry type sends out information about the user interface settings regarding the Planets Compare sonification. The OSC messages from this telemetry type goes under the OSC label `/Compare` and contains three items, explained in detail below:

>   1. The index of the first planet to be compared, see the first table below on how to convert the index to a planet name.

Author: Many things regarding the specialized telemetry types that are related to the sonification are fragile. This is because SuperCollider needs this to look (almost) exactly like this to work. Also, SuperCollider is a rather slow language, so we would like to avoid string comparisons (when possible) to make it a little bit faster. Especially since this is the most common message the sonification receives and uses.

| 1                      | Venus                                     |
| 2                      | Earth                                     |
| 3                      | Mars                                      |
| 4                      | Jupiter                                   |
| 5                      | Saturn                                    |
| 6                      | Uranus                                    |
| 7                      | Neptune                                   |
:::

Here is an example of how a message from this telemetry type can look:
:::{code-block}
[ /Overview, Int8Array[ 0, 0, 1, 0, 1, 1, 1, 1 ] ]
:::

Here is a breakdown of the example message above:
- `/Overview`: The OSC label indicating that this is a message from the **Planets Overview Sonification** telemetry type.
- `Int8Array[ 0, 0, 1, 0, 1, 1, 1, 1 ]`: The list of user interface settings, where:
    - `0` Mercury, is turned off.
    - `0` Venus, is turned off.
    - `1` Earth, is turned on.
    - `0` Mars, is turned off.
    - `1` Jupiter, is turned on.
    - `1` Saturn, is turned on.
    - `1` Uranus, is turned on.
    - `1` Neptune, is turned on.

# Sonification

> There is a sonification of the planets in our solar system that can be used together with the telemetry module. Sonification is when information is conveyed using sound in contrast to visualization when images are used. The files for the sonification can be found in _data\assets\modules\telemetry\sonification_ in the OpenSpace folder. The file _planets.asset_ is a regular Lua script OpenSpace asset that configures the telemetry module to monitor each of the planets in the solar system and their major moons. The sonification file itelf _OpenSpaceSonification.scd_ is a SuperCollider file and needs to be run seperatly in that program. The sonification will then be produced by SuperCollider and just uses the informaiton that OpenSpace sends with the OSC messages provided by the telemetry module.

Reviewer: ```suggestion
The [Telemetry Module](index.md) can be used together with a _sonification_ of the planets in our solar system. Sonification is the concept of conveying information using sound (in contrast to visualization, where information is conveyed using images). In addition to the telemetry module, the sonification requires a separate application/program called [SuperCollider](#supercollider). This software will be used as the receiver for the OSC messages from the telemetry module.

The files for the sonification can be found in _data\assets\modules\telemetry\sonification_ in the OpenSpace folder. The file _planets.asset_ is a regular OpenSpace asset that configures the telemetry module to monitor each of the planets in the solar system and their major moons. The sonification file itself is the _OpenSpaceSonification.scd_ file, which is a SuperCollider file and needs to be run separately in that program. The sonification will then be produced by SuperCollider and just uses the information that OpenSpace sends with the OSC messages provided by the telemetry module.
```

:::{note}

> Note that the _planets.asset_ file cannot be customized or altered since the sonification expects the data from the telemetry module to be in a certain format determined by this file. If this file is altered, then the sonification file needs to be updated to reflect the change as well. If you want to create your own sonification for other objects in OpenSpace other than the planets, then the general [Customized Nodes Information](./general.md#customized-nodes-information) telemetry type is likely a better fit for that purpose.

Reviewer: ```suggestion
Note that the _planets.asset_ file cannot be customized or altered since the sonification expects the data from the telemetry module to be in a certain format determined by this file. If this file is altered, then the sonification file needs to be updated to reflect the change as well. If you want to create your own sonification for objects in OpenSpace other than the planets, then the general [Customized Nodes Information](./general.md#customized-nodes-information) telemetry type is likely a better fit for that purpose.
```

:::

## SuperCollider

> [SuperCollider](https://supercollider.github.io/) is the software used to produce the sounds of the sonification. The messages from the telemetry module is sent to SuperCollider and it then uses that information to create sounds that corresonds to that informaiton. In order to run the sonification follow the steps below:

Reviewer: ```suggestion
[SuperCollider](https://supercollider.github.io/) is the software used to produce the sounds of the sonification. The messages from the telemetry module are sent to SuperCollider and it then uses that information to create sounds that respond to that information. To run the sonification follow the steps below:
```

3. Run OpenSpace with the profile called _sonification_
4. Turn on the Telemetry module in the user interface under _Settings/Modules_
5. Turn on the telemetries/sonifications that are of interest with the checkboxes in the telemetry module settings
6. Fly around in OpenSpace, and you should see messages being printed in the SuperCollider console window
7. Now, you can freely try the different settings for the different telemetries/sonifications, read the tooltips, and stress-test it in general.

:::{danger}
Before you shut down OpenSpace after using the telemetry module, you will need to turn off the telemetry module in the settings menu. If OpenSpace is shut down while the telemetry module is still enabled, it will **crash**. This is an issue with the Scene not being safe for multiple threads, especially around startup and shutdown. Due to the same reason, the telemetry module cannot be enabled with scripts before OpenSpace has properly booted up completely.
@TODO I am planning to (hopefully) fix the shutdown issue in th PR. However, the startup problem will remain.
:::

## The Sounds of the Planets Sonification

> The table below describes shortly what aspects of the planets are conveyed by the sonification, how they sound, and how the sounds change depending on the data.

Reviewer: ```suggestion
The table below describes what aspects of the planets are conveyed by the sonification, how they sound, and how the sounds change depending on the data.
```
